packages/svelte/src/legacy/legacy-client.js

```diff
@@ -71,50 +71,37 @@ class Svelte4Component {
 	 */
 	constructor(options) {
 		var sources = new Map();
-		var add_source = (/** @type {string | symbol} */ key) => {
-			var s = mutable_source(0);
+
+		/**
+		 * @param {string | symbol} key
+		 * @param {unknown} value
+		 */
+		var add_source = (key, value) => {
+			var s = mutable_source(value);
 			sources.set(key, s);
 			return s;
 		};
+
 		// Replicate coarse-grained props through a proxy that has a version source for
 		// each property, which is increment on updates to the property itself. Do not
 		// use our $state proxy because that one has fine-grained reactivity.
 		const props = new Proxy(
 			{ ...(options.props || {}), $$events: {} },
 			{
-				get(target, prop, receiver) {
-					var value = Reflect.get(target, prop, receiver);
-					var s = sources.get(prop);
-					if (s === undefined) {
-						s = add_source(prop);
-					}
-					get(s);
-					return value;
+				get(target, prop) {
+					return get(sources.get(prop) ?? add_source(prop, Reflect.get(target, prop)));
 				},
 				has(target, prop) {
-					var value = Reflect.has(target, prop);
-					var s = sources.get(prop);
-					if (s !== undefined) {
-						get(s);
-					}
-					return value;
+					get(sources.get(prop) ?? add_source(prop, Reflect.get(target, prop)));
```

Comment:
This isn't right. We don't want to add a source for a key that doesn't exist.

Reply:
Are you sure? It's buggy at present (though I didn't add a test already, because I didn't realise the behaviour was different until you pointed this out):

- [main](https://svelte-5-preview.vercel.app/#H4sIAAAAAAAAE32SMWvDMBCF_8qhBuyAsbN0cR1DydytW9PBtc-NQJaEdE4bhP57ZVtNQhM6aLin7969Q3Ks5wItK98ck82ArGTPWrOM0UlPhT2iIAy1VaNpJ6WyreGa6r3cEx-0MgS7Axcd9EYNkOTFXOVLY_J0hTlQ8kWNksBH-C7UGmwId6KxdqeCKPGmoxD42bSnpXE6Agm-TKM1mkkLShyVpmvY1uBmkFolLUF7dt3eHZZGeuKjVC47Zr8X1JhPDGqcuch-HWfvySK98gHV-CfAlWW-ClDqoFeqhOSAQqgE_DpaZfC42Vz8DNJoJCxeVxYdWjLqlM7kEqAqLg8kq44f4YPLrqQDt1sX8_q6KsJNHd51UB3vOXasJDOiz87fYF74v4_gHngPSYifAJewWmmjtPVTjkrX81YuinmofFXoEMmVKCxeUdAptCAVAX5zS5EqeO9vw737H7w8zj6vAgAA)
- [this PR](https://svelte-5-preview-git-simplify-legacy-props-svelte.vercel.app/#H4sIAAAAAAAAE32SMWvDMBCF_8qhBuyAsbN0cR1DydytW9PBtc-NQJaEdE4bhP57ZVtNQhM6aLin7969Q3Ks5wItK98ck82ArGTPWrOM0UlPhT2iIAy1VaNpJ6WyreGa6r3cEx-0MgS7Axcd9EYNkOTFXOVLY_J0hTlQ8kWNksBH-C7UGmwId6KxdqeCKPGmoxD42bSnpXE6Agm-TKM1mkkLShyVpmvY1uBmkFolLUF7dt3eHZZGeuKjVC47Zr8X1JhPDGqcuch-HWfvySK98gHV-CfAlWW-ClDqoFeqhOSAQqgE_DpaZfC42Vz8DNJoJCxeVxYdWjLqlM7kEqAqLg8kq44f4YPLrqQDt1sX8_q6KsJNHd51UB3vOXasJDOiz87fYF74v4_gHngPSYifAJewWmmjtPVTjkrX81YuinmofFXoEMmVKCxeUdAptCAVAX5zS5EqeO9vw737H7w8zj6vAgAA)

If we wanted to be super anal about preventing memory leaks then we could do the same thing we do with `SvelteSet` — have a `version` source that increments when new properties are added — but it feels unnecessary in this situation since the props you'd be testing for are finite.

Reply:
I do worry, but maybe less so given this is legacy. If this wasn't legacy, then yeah, we would need versions too.

Reply:
You should add that test BTW, we'll definitely need that to help future refactors.

Reply:
added the test

Reply:
Thanks